tsl/src/continuous_aggs/materialize.c

```diff
@@ -197,8 +198,21 @@ continuous_agg_materialize(int32 materialization_id, ContinuousAggMatOptions *op
 
 	catalog = ts_catalog_get();
 
-	/* copy over all the materializations from the raw hypertable to all the continuous aggs */
+	/* copy over all the materializations from the raw hypertable to all the continuous aggs
+	 */
 	caggs = ts_continuous_aggs_find_by_raw_table_id(cagg_data.raw_hypertable_id);
+	if (list_length(caggs) > 1)
+	{
+		/* REFRESH cagg_1; REFRESH cagg_2 can try to read+delete the
+		 * invalidation logs for the same hypertable.
+		 * If both REFRESH jobs try to delete the
+		 * same set of rows, we run into "tuple concurrently deleted error".
```

Comment:
I guess we could in theory handle this without locking. A concurrently deleted tuple might not strictly be an error  since it doesn't really matter which job deleted the tuple. If the job that experience the error aimed to delete the tuple itself, then I guess it is fine for that job to proceed as if it actually succeeded. 

This can, of course, be left as a future optimization. But I still figured it is worth pointing this out. Maybe leave a comment.